### ISSUE TITLE
P-Channel MOSFET Selection

### ISSUE THREAD

**Issue by mister-walter** [gregdavill/GlassUnicorn#3]
I'm trying to build one of your LED cubes, and was wondering if you could give any more information on the P-Channel MOSFETs that you used. I don't know a whole lot about component selection so I'm kind of flying by the seat of my pants here - I wasn't able to find any information about a P-Channel MOSFET with a designation 3412 (this is what appears in the BOM).

I'm mainly unsure about what max current capacity I should select for - the datasheet for the 2020 RGB LEDs that you linked on your blog indicates 20mA max forward current (I assume total between all color channels?), so my back-of-the-envelope calculation is 24 LEDs per row * 20mA = 0.5ish A. Therefore I should be looking for something with `Id > 1` and `Pd > 1.125W` (0.5A at 4.5V), right?

Thanks for posting this project!

**Comment by gregdavill** [comment]
The parts are from LCSC, but it appears they no longer sell them, and they've removed the listing.

From the invoice I have when I bought them. They have the following description:
`3415, Shenzhen Fuman Elec, P-Channel 12V 4A 1.4V @ 250uA 55mΩ @ 3A,4.5V MOSFET `

**Comment by mister-walter** [comment]
Thanks! I was able to find a similar MOSFET on LCSC.

Unrelated question - how did you select the resistor values for the TLC59025 clones? I'm planning to use an MBI5024 and I think I understand how to calculate the required resistor for a desired output current, but how did you determine the output currents for the 3 channels/what output current did you end up selecting for each?

**Comment by gregdavill** [comment]
I used the values from an off the shelf 64x64 LED panel. But it hod different controller ICs. So my white balance isn't great.

I think my currents are all around the 2-5mA range. I opted to go a bit lower to get better dynamic range. and I didn't need to blind anyone.

I'd recommend building up a board, and maybe trying to fit a trim-pot for each channel, and then with actual content displayed on the panel adjust for the best looking image, then use those values for the rest of the panels.
